core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/helper/DefaultLocalDcHelper.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.loadbalancing.helper;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.datastax.oss.driver.internal.core.metadata.DefaultNode;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Optional;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultLocalDcHelper extends BasicLocalDcHelper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultLocalDcHelper.class);
+
+  public DefaultLocalDcHelper(
+      @NonNull InternalDriverContext context,
+      @NonNull DriverExecutionProfile profile,
+      @NonNull String logPrefix) {
+    super(context, profile, logPrefix);
+  }
+
+  /**
+   * This implementation fetches the user-supplied datacenter, if any, from the programmatic
+   * configuration API, or else, from the driver configuration. If no local datacenter is explicitly
+   * defined, this implementation will consider two distinct situations:
+   *
+   * <ol>
+   *   <li>If no explicit contact points were provided, this implementation will infer the local
+   *       datacenter from the implicit contact point (localhost).
+   *   <li>If explicit contact points were provided however, this implementation will throw {@link
+   *       IllegalStateException}.
+   * </ol>
+   *
+   * @return The local datacenter; always present.
+   * @throws IllegalStateException if the local datacenter could not be inferred.
+   */
+  @NonNull
+  @Override
+  public Optional<String> discoverLocalDc() {
+    Optional<String> optionalLocalDc = super.discoverLocalDc();
+    if (optionalLocalDc.isPresent()) {
+      return optionalLocalDc;
+    }
+    Set<DefaultNode> contactPoints = context.getMetadataManager().getContactPoints();
+    if (context.getMetadataManager().wasImplicitContactPoint()) {
+      // We only allow automatic inference of the local DC in this specific case
+      assert contactPoints.size() == 1;
+      Node contactPoint = contactPoints.iterator().next();
+      String localDc = contactPoint.getDatacenter();
```

Comment:
how this edge case differ from `BasicLocalDcHelper`?  I remember from the discussion that `BasicLocalDcHelper` is used for the studio when they have only one contact point and they don't want to specify DC. For me, it looks that exactly this logic is covered in this `if`

Reply:
This LBP only infers the DC if no contact points were provided. This is never the case for Studio, where you need to specify at least (or maybe exactly, I don't know) one contact point.